docs/deploying.adoc

```diff
@@ -9,11 +9,250 @@ Put the filter jars in some directory
+broker address information like this:
+
+0.  `mybroker-0.mycluster.kafka.com:9193`
+1.  `mybroker-1.mycluster.kafka.com:9194`
+2.  `mybroker-2.mycluster.kafka.com:9194`
+
+NOTE: It is a responsible for the deployer of Kroxylicious to ensure that the bootstrap and generated broker
+DNS names are resolvable and routable by the Kafka Client.
+
+The `numberOfBrokerPorts` imposes a maximum on the number of brokers that a Kafka Cluster can have. Set this value
+to be as high as the maximum number of brokers that your operational rules allow for a Kafka Cluster.
+
+==== SniRouting scheme
+
+In the `SniRouting` scheme, Kroxylicious uses SNI information to route traffic to either the boostrap or individual
+brokers.   As this relies on SNI, the use of <<Downstream TLS>> is *required*.
+
+With this scheme, you have the choice to share a single port across all virtual clusters, or you can assign a different
+port to each.  Single port operation may have cost advantages when using load balancers of public clouds, as it allows
+a single cloud provider load balancer to be shared across all virtual clusters.
+
+[source, yaml]
+----
+clusterNetworkAddressConfigProvider:
+  type: SniRouting
+  config:
+    bootstrapAddress: mycluster.kafka.com:9192                    # <1>
+    brokerAddressPattern: mybroker-$(nodeId).mycluster.kafka.com  # <2>
+    bindAddress: 192.168.0.1                                      # <3>
+----
+<1> The hostname and port of the bootstrap that will be used by the Kafka Clients.
+<2> The broker address pattern used to form the broker addresses. The `$(nodeId)` token must be present.
+<3> (Optional) The bind address used when binding the port. If undefined, all network interfaces will be bound.
+
+The `brokerAddressPattern` configuration parameters requires that the `$(nodeId)` token is present within it.
+This is replaced by the https://kafka.apache.org/documentation/#brokerconfigs_node.id[`node.id`] (or `broker.id)
+assigned to the broker of the target cluster.  This allows Kroxylicious to generate separate routes for each broker.
+
+NOTE: It is a responsible for the deployer of Kroxylicious to ensure that the bootstrap and generated broker
+DNS names are resolvable and routable by the Kafka Client.
+
+=== Transport Layer Security (TLS)
+
+In this section we look at how to enable TLS for either the downstream and/or upstream.   Note, there is no
+interdependency; it is supported to have TLS configured for the downstream and use plain communications for the
+upstream, or vice-versa.
+
+NOTE: TLS is recommended for both upstream and downstream for production configurations.
+
+=== Downstream TLS
+
+Here's how to enable TLS for the downstream side. This means the Kafka Client will connect to the virtual cluster over
+TLS rather than plain.  For this, you will need to obtain a TLS certificate for the virtual cluster from your
```

Comment:
Guided by https://datatracker.ietf.org/doc/html/rfc4949, I think clear text is the right choice.

```
$ clear text
      1. (I) /noun/ Data in which the semantic information content
      (i.e., the meaning) is intelligible or is directly available,
      i.e., not encrypted. (See: cleartext, in the clear. Compare:
      cipher text, plain text.)

      2. (O) /noun/ "Intelligible data, the semantic content of which is
      available." [[I7498-2](https://datatracker.ietf.org/doc/html/rfc4949#ref-I7498-2)]

      3. (D) /noun/ Synonym for "plain text".

      Deprecated Definition: IDOCs SHOULD NOT use this term as a synonym
      for "plain text", because the plain text that is input to an
      encryption operation may itself be cipher text that was output
      from a previous encryption operation. (See: superencryption.)
```